service/lib/agama/storage/configs/sort_criteria.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2025] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Agama
+  module Storage
+    module Configs
+      # Namespace for criteria used to sort the result when searching devices.
+      module SortCriteria
+        # Base class for all sorting criteria
+        class Base
+          # Constructor
+          def initialize
+            @asc = true
+          end
+
+          # @return [Boolean] Whether sorting must be in ascending order
+          attr_accessor :asc
+          alias_method :asc?, :asc
+
+          # @return [Boolean] Whether sorting must be in descending order
```

Comment:
NP: I would simplify the API by removing `#desc?` and `#desc=`, or at least `#desc=`. This could be misleading:

~~~
criterion = Size.new
criterion.asc = true
criterion.desc = true
~~~